TFNetwork.py

```diff
@@ -1967,6 +1972,57 @@ def copy_new_base(self, name=None, layer=None, network=None, reduce_func=None):
       norm_factor=self._norm_factor, only_on_eval=self._only_on_eval)
 
 
+class MetaLossHolder:
+  """
+  Combines multiple losses in a customizable way.
+  It can be eg. used to compute the average over multiple losses which may then control the learning rate.
+  Depending on the way the losses are combined, the result may not be normalized.
+  This class supports only a subset of the functions in LossHolder. It is not supposed to be used eg. to compute
+  the objective function.
+  """
+  def __init__(self, losses_dict, eval_string):
+    """
+    :param dict[str,TFNetworkLayer.Loss] losses_dict: Mapping from loss names to losses.
+    :param string eval_string: Formula to combine the relevant losses. Will be eval'ed.
+    """
+    self.eval_string = eval_string
+
+    import re
+    term_pattern = re.compile('([a-zA-Z0-9_-]+)')
```

Comment:
Why the last `-`?

Reply:
If `-` is the last character of the regex, it is treated as a regular character, not a "from-to sign". And `-` could be a part of a valid layer name, right? So we need to include it here.

Reply:
No, I think it can not be part of a layer name (or should not).
Anyway, this is not a good idea, as a string like "a-b" becomes ambiguous (is that loss a - loss b or the layer with the name "a-b")